lib/apiservers/engine/backends/container.go

```diff
@@ -189,75 +188,18 @@ const (
 	defaultEnvPath = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 )
 
+// TODO: we should probably just make this purely a proxy thing. This is now just a passthrough
 func (c *Container) Handle(id, name string) (string, error) {
-	resp, err := c.containerProxy.Client().Containers.Get(containers.NewGetParamsWithContext(ctx).WithID(id))
-	if err != nil {
-		switch err := err.(type) {
-		case *containers.GetNotFound:
-			cache.ContainerCache().DeleteContainer(id)
-			return "", NotFoundError(name)
-		case *containers.GetDefault:
-			return "", InternalServerError(err.Payload.Message)
-		default:
-			return "", InternalServerError(err.Error())
-		}
-	}
-	return resp.Payload, nil
+	return c.containerProxy.Handle(id, name)
 }
 
 // docker's container.execBackend
 
-func (c *Container) TaskInspect(cid, cname, eid string) (*models.TaskInspectResponse, error) {
-	// obtain a portlayer client
-	client := c.containerProxy.Client()
-
-	handle, err := c.Handle(cid, cname)
-	if err != nil {
-		return nil, err
-	}
-
-	// inspect the Task to obtain ProcessConfig
-	config := &models.TaskInspectConfig{
-		Handle: handle,
-		ID:     eid,
-	}
-
-	params := tasks.NewInspectParamsWithContext(ctx).WithConfig(config)
-	resp, err := client.Tasks.Inspect(params)
-	if err != nil {
-		return nil, err
-	}
-	return resp.Payload, nil
-
-}
-
 func (c *Container) TaskWaitToStart(cid, cname, eid string) error {
-	// obtain a portlayer client
-	client := c.containerProxy.Client()
-
-	handle, err := c.Handle(cid, cname)
-	if err != nil {
-		return err
-	}
-
-	// wait the Task to start
-	config := &models.TaskWaitConfig{
-		Handle: handle,
-		ID:     eid,
-	}
-
-	params := tasks.NewWaitParamsWithContext(ctx).WithConfig(config)
-	_, err = client.Tasks.Wait(params)
-	if err != nil {
-		switch err := err.(type) {
-		case *tasks.WaitInternalServerError:
-			return InternalServerError(err.Payload.Message)
-		default:
-			return InternalServerError(err.Error())
-		}
-	}
+	op := trace.NewOperation(context.TODO(), "")
```

Comment:
This function isn't part of the backend container interface that we're implementing - don't know that it needs to be exported. It's also shallow enough that it could live only in the proxy.

Reply:
totally agreed, this will also tie the `opID` to the appropriate start operation. I am not sure as to whether or not that function is directly called by a docker client endpoint hit. I will check the configuration of the handlers to see if it is.